metrics/ci_integrations/test/source/github_actions/adapter/github_actions_source_client_adapter_test.dart

```diff
@@ -301,11 +331,13 @@ void main() {
           withJobsPage: defaultJobsPage,
         ).thenSuccessWith(workflowRunsPage);
 
-        final result = adapter.fetchBuilds(jobName);
+        final result = adapter.fetchBuilds(jobName, expectedNumberOfBuilds);
 
         expect(
           result,
-          completion(hasLength(GithubActionsSourceClientAdapter.fetchLimit)),
+          completion(
+            hasLength(expectedNumberOfBuilds),
+          ),
```

Comment:
```suggestion
          completion(
            hasLength(lessThanOrEqualTo(expectedNumberOfBuilds)),
          ),
```
Please, go through all similar tests and apply this suggestion.